### ISSUE TITLE
Improve test coverage

### ISSUE THREAD

**Issue by curbengh** [hexojs/hexo#3984]
## Check List

Please check followings before submitting a new feature request.

- [x] I have already read [Docs page](https://hexo.io/docs/)
- [x] I have already searched existing issues

## Feature Request

This repo's unit test coverage currently stands at [![Coverage Status](https://coveralls.io/repos/hexojs/hexo/badge.svg?branch=master)](https://coveralls.io/r/hexojs/hexo?branch=master).

For anyone who is interested in improving it, easiest way to start is to search for ["missing-unit-test"](https://github.com/hexojs/hexo/search?q=missing-unit-test&type=Code) to locate functions that haven't been covered. Otherwise, you could also click on the badge link and it shows all the relevant lines.

[hexo-util](https://github.com/hexojs/hexo-util)  [![Coverage Status](https://coveralls.io/repos/hexojs/hexo-util/badge.svg?branch=master)](https://coveralls.io/r/hexojs/hexo-util?branch=master) also needs help.

**Comment by seaoak** [comment]
@curbengh Thank you for nice waking-up call!

I checked a "missing-unit-test" point of `hexo-util`.
https://github.com/hexojs/hexo-util/blob/master/test/highlight.spec.js#L172

I found that `highlight()` of `hexo-util` might not work well for illegal syntax codes.
`highlight()` of `highlightjs` package will throw an exception if source text is illegal syntax codes.
https://highlightjs.readthedocs.io/en/latest/api.html#highlight-languagename-code-ignore-illegals-continuation

But `hexo-util/lib/highlight.js` might not consider that case.

I continue to investigate this issue (and make new issue in the repo of `hexo-util`).

**Comment by SukkaW** [comment]
After #4422, the coverage has been increased from 97.03% to 98.26%.

**Comment by stevenjoezhang** [comment]
```
npx nyc report --reporter=lcov --reporter=text
```

File                               | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s           
-----------------------------------|---------|----------|---------|---------|-----------------------------
All files                          |    97.3 |    92.55 |   98.22 |   98.32 |                             
 box                               |   92.81 |    83.64 |   96.23 |   94.78 |                             
  file.js                          |     100 |      100 |     100 |     100 |                             
  index.js                         |   92.09 |    83.64 |   95.83 |   94.17 | 155,230-231,235-236,249-250 
 extend                            |   97.55 |    93.69 |     100 |   99.06 |                             
  console.js                       |     100 |      100 |     100 |     100 |                             
  deployer.js                      |     100 |      100 |     100 |     100 |                             
  filter.js                        |   98.25 |    95.24 |     100 |     100 | 51-55                       
  generator.js                     |     100 |      100 |     100 |     100 |                             
  helper.js                        |     100 |      100 |     100 |     100 |                             
  index.js                         |     100 |      100 |     100 |     100 |                             
  injector.js                      |   94.74 |    92.16 |     100 |     100 | 53-58,64-69                 
  migrator.js                      |     100 |      100 |     100 |     100 |                             
  processor.js                     |   92.86 |    83.33 |     100 |   92.86 | 26                          
  renderer.js                      |   97.06 |    95.83 |     100 |     100 | 7                           
  tag.js                           |   97.37 |    89.36 |     100 |    98.1 | 228-229                     
 hexo                              |    97.9 |     92.8 |   98.41 |   98.83 |                             
  default_config.js                |     100 |      100 |     100 |     100 |                             
  index.js                         |   98.44 |    88.71 |   98.46 |   99.17 | 154,336                     
  load_config.js                   |   98.15 |       95 |     100 |     100 | 51                          
  load_database.js                 |     100 |      100 |     100 |     100 |                             
  load_plugins.js                  |   97.73 |    86.36 |   94.12 |   97.37 | 86                          
  load_theme_config.js             |     100 |      100 |     100 |     100 |                             
  locals.js                        |     100 |      100 |     100 |     100 |                             
  multi_config_path.js             |   97.78 |       95 |     100 |   97.67 | 57                          
  post.js                          |   97.37 |    93.24 |     100 |   99.26 | 54                          
  register_models.js               |     100 |      100 |     100 |     100 |                             
  render.js                        |   91.43 |    86.79 |     100 |   95.08 | 21-25                       
  router.js                        |   98.73 |    95.65 |   93.75 |   98.57 | 41                          
  scaffold.js                      |     100 |      100 |     100 |     100 |                             
  source.js                        |     100 |      100 |     100 |     100 |                             
  update_package.js                |     100 |      100 |     100 |     100 |                             
  validate_config.js               |     100 |      100 |     100 |     100 |                             
 models                            |   98.13 |    84.42 |     100 |   99.17 |                             
  asset.js                         |     100 |      100 |     100 |     100 |                             
  cache.js                         |     100 |      100 |     100 |     100 |                             
  category.js                      |   95.45 |    77.78 |     100 |     100 | 24,34-61                    
  data.js                          |     100 |      100 |     100 |     100 |                             
  index.js                         |     100 |      100 |     100 |     100 |                             
  page.js                          |     100 |      100 |     100 |     100 |                             
  post.js                          |   98.13 |    87.88 |     100 |   97.85 | 93,167                      
  post_asset.js                    |     100 |      100 |     100 |     100 |                             
  post_category.js                 |     100 |      100 |     100 |     100 |                             
  post_tag.js                      |     100 |      100 |     100 |     100 |                             
  tag.js                           |   97.44 |    71.43 |     100 |     100 | 13,18-52                    
 models/types                      |   95.45 |    94.29 |     100 |     100 |                             
  moment.js                        |   95.45 |    94.29 |     100 |     100 | 73-78                       
 plugins/console                   |    97.3 |    91.23 |     100 |    98.5 |                             
  clean.js                         |     100 |      100 |     100 |     100 |                             
  config.js                        |     100 |       95 |     100 |     100 | 19                          
  deploy.js                        |   97.22 |    92.86 |     100 |     100 | 42                          
  generate.js                      |    95.8 |    88.64 |     100 |   97.17 | 102,136,144                 
  index.js                         |     100 |      100 |     100 |     100 |                             
  migrate.js                       |     100 |      100 |     100 |     100 |                             
  new.js                           |   93.75 |    91.67 |     100 |   92.31 | 26                          
  publish.js                       |     100 |      100 |     100 |     100 |                             
  render.js                        |   95.83 |       80 |     100 |     100 | 25,33                       
 plugins/console/list              |   73.58 |       65 |   73.68 |   74.44 |                             
  category.js                      |     100 |      100 |     100 |     100 |                             
  common.js                        |      90 |       50 |     100 |   88.89 | 14                          
  index.js                         |   77.78 |       75 |      50 |    87.5 | 23                          
  page.js                          |     100 |      100 |     100 |     100 |                             
  post.js                          |   94.44 |       75 |      75 |   93.75 | 8                           
  route.js                         |    7.69 |        0 |       0 |    9.09 | 6-50                        
  tag.js                           |     100 |      100 |     100 |     100 |                             
 plugins/filter                    |     100 |      100 |     100 |     100 |                             
  index.js                         |     100 |      100 |     100 |     100 |                             
  new_post_path.js                 |     100 |      100 |     100 |     100 |                             
  post_permalink.js                |     100 |      100 |     100 |     100 |                             
 plugins/filter/after_post_render  |     100 |      100 |     100 |     100 |                             
  excerpt.js                       |     100 |      100 |     100 |     100 |                             
  external_link.js                 |     100 |      100 |     100 |     100 |                             
  index.js                         |     100 |      100 |     100 |     100 |                             
 plugins/filter/after_render       |   97.06 |    95.45 |     100 |     100 |                             
  external_link.js                 |   95.24 |    92.86 |     100 |     100 | 12                          
  index.js                         |     100 |      100 |     100 |     100 |                             
  meta_generator.js                |     100 |      100 |     100 |     100 |                             
 plugins/filter/before_exit        |     100 |      100 |     100 |     100 |                             
  index.js                         |     100 |      100 |     100 |     100 |                             
  save_database.js                 |     100 |      100 |     100 |     100 |                             
 plugins/filter/before_generate    |     100 |      100 |     100 |     100 |                             
  index.js                         |     100 |      100 |     100 |     100 |                             
  render_post.js                   |     100 |      100 |     100 |     100 |                             
 plugins/filter/before_post_render |     100 |       92 |     100 |     100 |                             
  backtick_code_block.js           |     100 |    93.18 |     100 |     100 | 32,65,87                    
  index.js                         |     100 |      100 |     100 |     100 |                             
  titlecase.js                     |     100 |    83.33 |     100 |     100 | 8                           
 plugins/filter/template_locals    |     100 |      100 |     100 |     100 |                             
  i18n.js                          |     100 |      100 |     100 |     100 |                             
  index.js                         |     100 |      100 |     100 |     100 |                             
 plugins/generator                 |   98.18 |      100 |   92.86 |   97.87 |                             
  asset.js                         |   95.83 |      100 |   88.89 |      95 | 28                          
  index.js                         |     100 |      100 |     100 |     100 |                             
  page.js                          |     100 |      100 |     100 |     100 |                             
  post.js                          |     100 |      100 |     100 |     100 |                             
 plugins/helper                    |   99.17 |    94.81 |   98.89 |   99.87 |                             
  css.js                           |     100 |      100 |     100 |     100 |                             
  date.js                          |     100 |    97.14 |     100 |     100 | 14                          
  debug.js                         |     100 |      100 |     100 |     100 |                             
  favicon_tag.js                   |     100 |      100 |     100 |     100 |                             
  feed_tag.js                      |     100 |      100 |     100 |     100 |                             
  format.js                        |      90 |      100 |       0 |     100 |                             
  fragment_cache.js                |     100 |      100 |     100 |     100 |                             
  full_url_for.js                  |     100 |      100 |     100 |     100 |                             
  gravatar.js                      |     100 |      100 |     100 |     100 |                             
  image_tag.js                     |     100 |      100 |     100 |     100 |                             
  index.js                         |     100 |      100 |     100 |     100 |                             
  is.js                            |     100 |      100 |     100 |     100 |                             
  js.js                            |     100 |      100 |     100 |     100 |                             
  link_to.js                       |     100 |      100 |     100 |     100 |                             
  list_archives.js                 |   98.57 |    95.83 |     100 |     100 | 28,44                       
  list_categories.js               |   98.44 |    91.55 |     100 |     100 | 11,55,87-95                 
  list_posts.js                    |     100 |     87.1 |     100 |     100 | 23,31,47-50                 
  list_tags.js                     |   98.08 |    93.75 |     100 |     100 | 11,72-74                    
  mail_to.js                       |     100 |      100 |     100 |     100 |                             
  markdown.js                      |     100 |      100 |     100 |     100 |                             
  meta_generator.js                |     100 |      100 |     100 |     100 |                             
  number_format.js                 |     100 |    86.96 |     100 |     100 | 11-16,33                    
  open_graph.js                    |   99.03 |    94.35 |     100 |   98.95 | 45                          
  paginator.js                     |     100 |    91.49 |     100 |     100 | 62-72,89,94                 
  partial.js                       |     100 |      100 |     100 |     100 |                             
  relative_url.js                  |     100 |      100 |     100 |     100 |                             
  render.js                        |     100 |      100 |     100 |     100 |                             
  search_form.js                   |     100 |      100 |     100 |     100 |                             
  tagcloud.js                      |   95.83 |    95.74 |     100 |     100 | 29,51                       
  toc.js                           |     100 |      100 |     100 |     100 |                             
  url_for.js                       |     100 |      100 |     100 |     100 |                             
 plugins/injector                  |     100 |      100 |     100 |     100 |                             
  index.js                         |     100 |      100 |     100 |     100 |                             
 plugins/processor                 |   94.85 |    88.06 |   96.88 |   95.88 |                             
  asset.js                         |   95.45 |    91.11 |     100 |   95.24 | 20,28,102                   
  common.js                        |     100 |      100 |     100 |     100 |                             
  data.js                          |   83.33 |       60 |     100 |    87.5 | 16,24                       
  index.js                         |     100 |      100 |     100 |     100 |                             
  post.js                          |   94.33 |    86.67 |   92.86 |      96 | 36,44,161-162,190           
 plugins/renderer                  |   98.25 |    95.83 |     100 |   98.18 |                             
  index.js                         |     100 |      100 |     100 |     100 |                             
  json.js                          |     100 |      100 |     100 |     100 |                             
  nunjucks.js                      |   97.14 |    95.83 |     100 |   96.97 | 24                          
  plain.js                         |     100 |      100 |     100 |     100 |                             
  yaml.js                          |     100 |      100 |     100 |     100 |                             
 plugins/tag                       |   99.46 |    94.33 |     100 |     100 |                             
  asset_img.js                     |     100 |      100 |     100 |     100 |                             
  asset_link.js                    |     100 |      100 |     100 |     100 |                             
  asset_path.js                    |     100 |      100 |     100 |     100 |                             
  blockquote.js                    |     100 |    90.91 |     100 |     100 | 27,39                       
  code.js                          |   98.72 |    92.31 |     100 |     100 | 59,70,114-115               
  gist.js                          |     100 |      100 |     100 |     100 |                             
  iframe.js                        |     100 |      100 |     100 |     100 |                             
  img.js                           |     100 |    93.75 |     100 |     100 | 44                          
  include_code.js                  |   98.08 |    83.33 |     100 |     100 | 26-47,60-61                 
  index.js                         |     100 |      100 |     100 |     100 |                             
  jsfiddle.js                      |     100 |      100 |     100 |     100 |                             
  link.js                          |     100 |      100 |     100 |     100 |                             
  post_link.js                     |     100 |      100 |     100 |     100 |                             
  post_path.js                     |     100 |      100 |     100 |     100 |                             
  pullquote.js                     |     100 |      100 |     100 |     100 |                             
  vimeo.js                         |     100 |      100 |     100 |     100 |                             
  youtube.js                       |     100 |      100 |     100 |     100 |                             
 theme                             |   98.25 |       88 |     100 |     100 |                             
  index.js                         |   97.37 |       80 |     100 |     100 | 25,73                       
  view.js                          |   98.68 |       90 |     100 |     100 | 10,30,55,117                
 theme/processors                  |    96.3 |    89.47 |     100 |   98.04 |                             
  config.js                        |     100 |      100 |     100 |     100 |                             
  i18n.js                          |   93.33 |       75 |     100 |     100 | 18                          
  source.js                        |   94.74 |    90.91 |     100 |   94.12 | 17                          
  view.js                          |     100 |      100 |     100 |     100 |
